frontend/src/metabase/dashboard/containers/DashboardHeader.jsx

```diff
@@ -145,8 +145,8 @@ class DashboardHeader extends Component {
     );
   }
 
-  async onSave() {
-    await this.props.saveDashboardAndCards(this.props.dashboard.id);
+  async onSave(preserveParameters = false) {
```

Comment:
there are 2 places that call this function in this file
1. The **Save** button when clicking editing the dashboard and entering the editing mode. The behavior will remain the same as before.
2. When editing the dashboard name in the editor outside the editing mode, which will instantly update the dashboard name. In this case, we can tell `saveDashboardAndCards` we want to preserve parameter values since we're not touching parameters at all.

Reply:
nit: I'd omit default `false`, seems straightforward without the default value

Reply:
Agree, I already default the parameter to `false` in the action so it's actually optional. I will fix its type to say the boolean is optional as well.